editor/reducer.js

```diff
@@ -546,6 +546,65 @@ export function notices( state = {}, action ) {
 	return state;
 }
 
+export function reusableBlocks( state = {}, action ) {
+	switch ( action.type ) {
+		case 'FETCH_REUSABLE_BLOCKS_SUCCESS': {
+			return reduce( action.reusableBlocks, ( newState, reusableBlock ) => ( {
+				...newState,
+				[ reusableBlock.id ]: {
+					...reusableBlock,
+					isSaving: false,
```

Comment:
Why are we merging the `isSaving` flag with thee reusable block object, my first reaction here is: "Show this be in a separate reducer (or subReducer)?

Reply:
No reason other than that it's convenient to pass the block's data and metadata (`isSaving`) to UI components all in one go.

I'd be happy to move this to a different reducer. Perhaps the shape of the redux tree could be something like:
```js
reusableBlocks: {
    data: {
        'f5ca58b0-aad0-4777-bb10-f3d0fa08a163': /* the block */,
    },
    isSaving: {
        'f5ca58b0-aad0-4777-bb10-f3d0fa08a163': false,
    },
}
```

Reply:
> Perhaps the shape of the redux tree could be something like

I like this proposal. Request flags could be generic, it makes sense to me to separate them from the data.

Reply:
👌 done in fb9d4c70a898908eddcf9b71028d16262d171791.